LoopKit/AnyCodableEquatable.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  AnyCodableEquatable.swift
+//  LoopKit
+//
+//  Created by Darin Krauss on 2/8/22.
+//  Copyright © 2022 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+
+public struct AnyCodableEquatable: Codable, Equatable {
+    public enum Error: Swift.Error {
+        case unknownType
+    }
+
+    public let any: Any
+    private let equals: (Self) -> Bool
+
+    public init<T: Codable & Equatable>(_ any: T) {
+        self.any = any
+        self.equals = { $0.any as? T == any }
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        if let value = try? container.decode(String.self) {
```

Comment:
I thought you needed `Date`...did you not want to use that?

Reply:
Turns out I was storing a TimeInterval and not a Date so I removed this. If we need it later we can add it back.

Reply:
Essentially, I just went with the basic JSON data types (less object and array).